api/nakadi-event-bus-api.yaml

```diff
@@ -1083,6 +1305,67 @@ definitions:
         items:
           $ref: '#/definitions/Event'
 
+  Subscription:
+    description: |
+      Subscription is a high level consumption unit. Subscriptions allow applications to easily scale the
+      number of clients by managing consumed event offsets and distributing load between instances.
+      The key properties that identify subscription are 'owning_application', 'event_types' and 'consumer_group'.
+      It's not possible to have two different subscriptions with these properties being the same.
+    properties:
+      id:
+        type: string
+        format: uuid
+        readOnly: true
+        description: |
+          Id of subscription that was created. Is generated by Nakadi, should not be specified when creating
+          subscription.
+      owning_application:
+        type: string
+        example: 'gizig'
+        description: |
+          The id of application owning the subscription. The reading and modification of subscription will be limited
+          to this application id.
+      event_types:
+        type: array
+        items:
+          type: string
+        description: |
+          EventTypes to subscribe to.
+          The order is not important. Subscriptions that differ only be the order of EventTypes will be
+          considered the same and will have the same id.
+
+          * Currently only subscription to a single EventType is supported. Subscriptions with more than one
+            EventType in event_types property will be rejected.
+      consumer_group:
+        type: string
+        example: 'read-product-updates'
+        description: |
+          The value describing the use case of this subscription.
+          In general that is an additional identifier used to differ subscriptions having the same
+          owning_application and event_types.
+        default: 'none'
+      created_at:
+        type: string
+        readOnly: true
+        description: |
+          Timestamp of creation of the subscription. Is generated by Nakadi, should not be specified when
+          creating subscription.
+        format: date-time
+        example: '1996-12-19T16:39:57-08:00'
+      read_from:
+        type: string
+        enum:
+          - begin
+          - end
+        description: |
+          Position to start reading events from. "begin" - read from the oldest available event. "end" - read from the
+          most recent offset. Applied in the moment when client starts reading from a subscription.
+        default: 'end'
+
+    required:
+      - owning_application
+      - event_types
+
```

Comment:
I would agree if there were 3 variants of this object. Then your suggestion would make it easy to extend, for example, creating a `SubscriptionToUpdate` and so on. But since there are only 2 possible "states", I think the "ifs" doesn't harm.